### ISSUE TITLE
BUG: MultiIndex columns containing datetime cannot be used (regression from 1.0.5 to master)

### ISSUE THREAD

**Issue by Dr-Irv** [pandas-dev/pandas#35015]
- [x] I have checked that this issue has not already been reported.

- [ ] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
In [1]: import pandas as pd

In [2]: pd.__version__
Out[2]: '1.1.0.dev0+1967.gf5b2e5a9a'

In [3]: expected_column_index = pd.MultiIndex.from_tuples(
   ...:     [(pd.to_datetime("02/29/2020"), pd.to_datetime("03/01/2020"))],
   ...:     names=[
   ...:         'a', 'b'])

In [4]: pd.DataFrame([], columns=expected_column_index)
---------------------------------------------------------------------------
InvalidIndexError
```
#### Problem description

This works in 1.0.5, but not in master.

See discussion in #34748 .  This new issue should be resolved before that PR moves forward.

#### Expected Output

```python
In [1]: import pandas as pd

In [2]: pd.__version__
Out[2]: '1.0.5'

In [3]:  expected_column_index = pd.MultiIndex.from_tuples(
   ...:        [(pd.to_datetime("02/29/2020"), pd.to_datetime("03/01/2020"))],
   ...:        names=[
   ...:            'a', 'b'])

In [4]: pd.DataFrame([], columns=expected_column_index)
Out[4]:
Empty DataFrame
Columns: [(2020-02-29 00:00:00, 2020-03-01 00:00:00)]
Index: []
```

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit           : f5b2e5a9a2431165b1e8002219c5d3962d7bc920
python           : 3.7.6.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18362
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 13, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 1.1.0.dev0+1967.gf5b2e5a9a
numpy            : 1.18.5
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.1.1
setuptools       : 42.0.2.post20191201
Cython           : 0.29.20
pytest           : 5.4.3
hypothesis       : 5.18.0
sphinx           : 3.1.1
blosc            : None
feather          : None
xlsxwriter       : 1.2.9
lxml.etree       : 4.5.1
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.15.0
pandas_datareader: None
bs4              : 4.9.1
bottleneck       : 1.3.2
fsspec           : 0.7.4
fastparquet      : 0.4.0
gcsfs            : None
matplotlib       : 3.1.2
numexpr          : 2.7.1
odfpy            : None
openpyxl         : 3.0.3
pandas_gbq       : None
pyarrow          : 0.15.1
pytables         : None
pyxlsb           : 1.0.6
s3fs             : 0.4.2
scipy            : 1.3.1
sqlalchemy       : 1.3.17
tables           : 3.6.1
tabulate         : 0.8.7
xarray           : 0.15.1
xlrd             : 1.2.0
xlwt             : 1.3.0
numba            : 0.46.0
</details>

**Comment by Dr-Irv** [comment]
@jbrockmendel This error seems to be related to code you were working on with respect to the datetime index.  The error that is raised is due to a commit you did here:  https://github.com/pandas-dev/pandas/commit/0b0cd08524e4472eb15835c2b91621dc0a6eeeb0

Can you see if you can fix this and then I can update PR #34748 ?

**Comment by jbrockmendel** [comment]
@Dr-Irv looking at this, it appears that a sufficient fix is to add in core.indexing._LocIndexer._convert_to_indexer:

```
            try:
                return labels.get_loc(key)
            except LookupError:
                if isinstance(key, tuple) and isinstance(labels, ABCMultiIndex):
                    if len(key) == labels.nlevels:
                        return {"key": key}
                    raise
+            except InvalidIndexError:
+               if not isinstance(labels, ABCMultiIndex):
+                    raise
```

I haven't run the test suite with this, nor looked at this closely enough to confirm that its the "correct" fix, but it does fix the example in the OP.

**Comment by Dr-Irv** [comment]
@jbrockmendel Thanks.  I will create a test, try that fix and let you know if something else pops up

**Comment by Dr-Irv** [comment]
@jbrockmendel OK, that seemed to fix it in #35140 .  Thanks!
